### PR TITLE
Volodymyr mescherjak

### PR DESCRIPTION
**Pull request by merkan3k**
LVJAVS-458 - Error messages don't appear when enter too short name in ‘First name’ and ‘Last name’ fields.

Валідація проходить як на фронтенді  - до файлу edit-my-kid.jsp додано скріпти validation-edit-kid.js та validation-edit-kid - ua.js, що використовуються в залежності від локалізації;
                                        так і на бекенді - у файлі ChildValidator.java

### CODE REVIEW

**Comment by vplytan-amplify** [comment]
please, write PR description in English

File: src/main/java/ua/softserveinc/tc/service/impl/BookingServiceImpl.java [hunk 1]
```diff
@@ -164,18 +164,40 @@ public Boolean checkForDuplicateBookingSingle(BookingDto bookingDto) {
...
     @Override
     public List<BookingDto> persistBookingsFromDtoAndSetId(List<BookingDto> listDTO) {
-        BookingDto bdto = listDTO.get(0);
-
-        int available = roomService.getAvailableSpaceForPeriod(
-                bdto.getDateStartTime(),
-                bdto.getDateEndTime(),
-                bdto.getRoom());
+        if(listDTO == null)
+            throw new RuntimeException("listDTO is null");
```
Comment (vplytan-amplify) [inline src/main/java/ua/softserveinc/tc/service/impl/BookingServiceImpl.java]: RuntimeException ?
is is really that critical ?
okay, first of all you need to cover this with unit test
then make sure this list is nevew null
then you can avoid throwing exception - just handle this null case ;)

File: src/main/java/ua/softserveinc/tc/service/impl/BookingServiceImpl.java [hunk 2]
```diff
@@ -164,18 +164,40 @@ public Boolean checkForDuplicateBookingSingle(BookingDto bookingDto) {
...
+    private boolean isAvailabilePlacesInTheRoom(List<BookingDto> listDTO) {
+        int availablePlaces = 0;
+        int needPlaces = 1;
+        Date theSameDay = null;
+        for(BookingDto bdto : listDTO) {
+            if(bdto.getDateStartTime().equals(theSameDay)) {
+                needPlaces++;
+                continue;
+            } else if (theSameDay != null && availablePlaces < needPlaces)
+                return false;
```
Comment (vplytan-amplify) [inline src/main/java/ua/softserveinc/tc/service/impl/BookingServiceImpl.java]: avoid multiple returns

File: src/main/java/ua/softserveinc/tc/service/impl/RoomServiceImpl.java [hunk 3]
```diff
@@ -169,14 +169,47 @@ public Boolean isPossibleUpdate(BookingDto bookingDto) {
...
 
     /**
+     * The method finds the maximum people in the room for period of time
+     * from dateLo to dateHi.
+     * @param dateLo start of period
+     * @param dateHi end of period
+     * @param bookings all reserved bookings in the time period
+     * @return The maximum number of people that are simultaneously in the room
+     */
+
```
Comment (vplytan-amplify) [inline src/main/java/ua/softserveinc/tc/service/impl/RoomServiceImpl.java]: remove this line

File: src/main/java/ua/softserveinc/tc/service/impl/RoomServiceImpl.java [hunk 4]
```diff
@@ -169,14 +169,47 @@ public Boolean isPossibleUpdate(BookingDto bookingDto) {
...
+    private int maxRangeReservedBookings (Date dateLo, Date dateHi, List<Booking> bookings) {
+        if(dateLo == null)
+            throw new NullPointerException("dateLo is null");
+        else if(dateHi == null)
+            throw new NullPointerException("dateHi is null");
+        else if(bookings == null)
```
Comment (vplytan-amplify) [inline src/main/java/ua/softserveinc/tc/service/impl/RoomServiceImpl.java]: this is wrong
at least you can do all these checks in one if statement
and do not throw Exceptions - if you have some nulls - just handle them.

File: src/main/java/ua/softserveinc/tc/service/impl/RoomServiceImpl.java [hunk 5]
```diff
@@ -169,14 +169,47 @@ public Boolean isPossibleUpdate(BookingDto bookingDto) {
...
+            throw new NullPointerException("bookings is null");
+
+        long oneMinute = 60 * 1000;
+        int maxReservedBookings = 0;
+        for(long ti = dateLo.getTime(); ti < dateHi.getTime(); ti += oneMinute) {
+            int temporaryMax = 0;
+            for(Booking tab : bookings)
```
Comment (vplytan-amplify) [inline src/main/java/ua/softserveinc/tc/service/impl/RoomServiceImpl.java]: bracket ?

File: src/main/java/ua/softserveinc/tc/service/impl/RoomServiceImpl.java [hunk 6]
```diff
@@ -169,14 +169,47 @@ public Boolean isPossibleUpdate(BookingDto bookingDto) {
...
+                if(tab.getBookingStartTime().getTime() <= ti && tab.getBookingEndTime().getTime() >= ti)
+                    temporaryMax++;
+            if(temporaryMax > maxReservedBookings)
```
Comment (vplytan-amplify) [inline src/main/java/ua/softserveinc/tc/service/impl/RoomServiceImpl.java]: where are brackets to your if statements ?

File: src/main/java/ua/softserveinc/tc/service/impl/RoomServiceImpl.java [hunk 7]
```diff
@@ -169,14 +169,47 @@ public Boolean isPossibleUpdate(BookingDto bookingDto) {
...
+        if(dateLo == null)
+            throw new NullPointerException("dateLo is null");
+        else if(dateHi == null)
+            throw new NullPointerException("dateHi is null");
+        else if(bookings == null)
+            throw new NullPointerException("bookings is null");
+
+        long oneMinute = 60 * 1000;
+        int maxReservedBookings = 0;
+        for(long ti = dateLo.getTime(); ti < dateHi.getTime(); ti += oneMinute) {
```
Comment (vplytan-amplify) [inline src/main/java/ua/softserveinc/tc/service/impl/RoomServiceImpl.java]: what does this mean ? are you travelling in time here ?
this should be simplified

File: src/main/java/ua/softserveinc/tc/validator/ChildValidator.java [hunk 8]
```diff
@@ -38,6 +38,23 @@ public void validate(Object o, Errors errors) {
...
 
+
+/**
+ * check if the first and last names are corrent(if there is only 1 letter)
+ * Error messages appear. "First Name" and "Last Name" fields are not edited.
+ */
+        String fName = kidToValidate.getFirstName();
+        String lName = kidToValidate.getLastName();
+
+        if ((fName.length() < 2) || (lName.length() < 2)) {
```
Comment (vplytan-amplify) [inline src/main/java/ua/softserveinc/tc/validator/ChildValidator.java]: 2 - move to constant and javadoc that constant

Comment (vplytan-amplify) [inline src/main/java/ua/softserveinc/tc/validator/ChildValidator.java]: and why do you need this chech if you have your inner checks ?

File: src/main/webapp/WEB-INF/views/user/edit-my-kid.jsp [hunk 9]
```diff
@@ -21,14 +22,14 @@ modelAttribute="<%=ChildConstants.View.KID_ATTRIBUTE %>" action="editmykid" meth
       <form:hidden path="id" />
 <div class="form-group-material-blue-400">
       <div class="form-group">
-             <label for="firstname" class="required"><spring:message code="kid.firstname" /></label>
-              <form:input path="firstName" id="firstname" value="${kid.getFirstName()}"
+             <label for="firstName" class="required"><spring:message code="kid.firstname" /></label>
+              <form:input path="firstName" id="firstName" value="${kid.getFirstName()}"
```
Comment (vplytan-amplify) [inline src/main/webapp/WEB-INF/views/user/edit-my-kid.jsp]: formatting

File: src/main/webapp/resources/js/renderCalendar.js [hunk 10]
```diff
@@ -55,7 +55,7 @@ $(function () {
 
     $('.timepicker').timepicker({
         timeFormat: 'H:i',
-        step: 15,
+        step: 1,
```
Comment (vplytan-amplify) [inline src/main/webapp/resources/js/renderCalendar.js]: just wondering - why this change is required ?

File: src/main/webapp/resources/js/validation-edit-kid-ua.js [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+ * Created by Вова on 14.11.2016.
+ */
+$(document).ready(function() {
+    $(function() {
+        $.validator.addMethod("regexFirstName", function(value, element, regexpr) {
+            return regexpr.test(value);
+        }, "Все плохо");
+        $.validator.addMethod("regexLastName", function(value, element, regexpr) {
+            return regexpr.test(value);
+        }, "Все плохо");
```
Comment (vplytan-amplify) [inline src/main/webapp/resources/js/validation-edit-kid-ua.js]: what does it mean ?

File: src/main/webapp/resources/js/validation-edit-kid-ua.js [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+                    required: "Будь ласка введіть щонайменше 2 символи"
+
+                },
+                lastName: {
+                    required: "Будь ласка введіть щонайменше 2 символи"
+                }
+            }
+        });
+    });
+});
```
Comment (vplytan-amplify) [inline src/main/webapp/resources/js/validation-edit-kid-ua.js]: add newline in the end of the file

File: src/main/webapp/resources/js/validation-edit-kid.js [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+                },
+                lastName: {
+                    required: 'Please enter at least 2 characters'
+
+                }
+
+            }
+        });
+    });
+});
```
Comment (vplytan-amplify) [inline src/main/webapp/resources/js/validation-edit-kid.js]: new line

File: src/main/webapp/WEB-INF/views/user/edit-my-kid.jsp (hunk 9 above)
Comment (merkan3k) [inline src/main/webapp/WEB-INF/views/user/edit-my-kid.jsp]: It is not my code.

File: src/main/java/ua/softserveinc/tc/service/impl/RoomServiceImpl.java (hunk 5 above)
Comment (merkan3k) [inline src/main/java/ua/softserveinc/tc/service/impl/RoomServiceImpl.java]: it is not my code

File: src/main/java/ua/softserveinc/tc/service/impl/RoomServiceImpl.java (hunk 7 above)
Comment (merkan3k) [inline src/main/java/ua/softserveinc/tc/service/impl/RoomServiceImpl.java]: It's not my code

File: src/main/java/ua/softserveinc/tc/service/impl/RoomServiceImpl.java (hunk 4 above)
Comment (merkan3k) [inline src/main/java/ua/softserveinc/tc/service/impl/RoomServiceImpl.java]: It's not my code

File: src/main/java/ua/softserveinc/tc/service/impl/RoomServiceImpl.java (hunk 3 above)
Comment (merkan3k) [inline src/main/java/ua/softserveinc/tc/service/impl/RoomServiceImpl.java]: It's not my code

File: src/main/java/ua/softserveinc/tc/service/impl/BookingServiceImpl.java (hunk 1 above)
Comment (merkan3k) [inline src/main/java/ua/softserveinc/tc/service/impl/BookingServiceImpl.java]: It's not my code

File: src/main/java/ua/softserveinc/tc/service/impl/BookingServiceImpl.java (hunk 2 above)
Comment (merkan3k) [inline src/main/java/ua/softserveinc/tc/service/impl/BookingServiceImpl.java]: It's not my code

File: src/main/java/ua/softserveinc/tc/service/impl/RoomServiceImpl.java (hunk 6 above)
Comment (merkan3k) [inline src/main/java/ua/softserveinc/tc/service/impl/RoomServiceImpl.java]: It's not my code

File: src/main/java/ua/softserveinc/tc/validator/ChildValidator.java (hunk 8 above)
Comment (merkan3k) [inline src/main/java/ua/softserveinc/tc/validator/ChildValidator.java]: done

File: src/main/webapp/resources/js/renderCalendar.js (hunk 10 above)
Comment (merkan3k) [inline src/main/webapp/resources/js/renderCalendar.js]: It's not my code

File: src/main/webapp/resources/js/validation-edit-kid-ua.js (hunk 11 above)
Comment (merkan3k) [inline src/main/webapp/resources/js/validation-edit-kid-ua.js]: done

File: src/main/webapp/resources/js/validation-edit-kid-ua.js (hunk 12 above)
Comment (merkan3k) [inline src/main/webapp/resources/js/validation-edit-kid-ua.js]: done

File: src/main/webapp/resources/js/validation-edit-kid.js (hunk 13 above)
Comment (merkan3k) [inline src/main/webapp/resources/js/validation-edit-kid.js]: done
